saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -494,7 +504,71 @@ def trigger_webhooks_async(
     )
 
 
-@app.task(bind=True)
+def get_last_event_delivery_time(db_connection) -> datetime.datetime | None:
+    """Get the time of the last event delivery object present on the replica side."""
+    event_delivery = EventDelivery.objects.using(db_connection).order_by("-id").first()
+    if event_delivery:
+        return event_delivery.created_at
+    return None
+
+
+def is_replica_ready_for_payload_generation(
+    celery_task: Task,
+    payload_requested_at: datetime.datetime | None,
+    event_delivery_ids: list[int],
+    retry_backoff: int,
+    db_connection: str,
+):
+    """Check if the read replica is caught up enough to generate payload.
+
+    If up to date, allow the payload generation to proceed.
+    If not, retry the task until the replica is caught up.
+    In case of max retries exceeded, mark the deliveries as failed.
+    """
+    current_last_event_delivery_time = get_last_event_delivery_time(db_connection)
+
+    if (
+        current_last_event_delivery_time and payload_requested_at
+    ) and current_last_event_delivery_time < payload_requested_at:
+        logger.warning(
+            "Postponing deferred payload generation until replica is caught up. "
+            "Replica has last event delivery created at: %s, payload requested at: %s (diff %s sec). "
+            "Current attempt: %d/%d",
+            current_last_event_delivery_time,
+            payload_requested_at,
+            (payload_requested_at - current_last_event_delivery_time).total_seconds(),
+            celery_task.request.retries,
+            celery_task.max_retries,
+            extra={
+                "event_delivery_ids": event_delivery_ids,
+            },
+        )
+        try:
+            countdown = retry_backoff * (2**celery_task.request.retries)
+            task_properties = celery_task.request.properties or {}
+            message_group_id = task_properties.get("MessageGroupId")
+            raise celery_task.retry(
+                countdown=countdown,
+                # Set delay when using SQS transport
+                DelaySeconds=countdown,
```

Comment:
I think we should only set countdown here and modify the SQS transport to inject a `DelaySeconds` argument (capped to 900 if the delay is longer) if `eta` or `countdown` is set.